vector/src/main/java/im/vector/app/features/home/room/detail/search/SearchViewState.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.home.room.detail.search
+
+import com.airbnb.mvrx.MvRxState
+import org.matrix.android.sdk.api.session.search.SearchResult
+
+data class SearchViewState(
+        // Accumulated search result
+        val searchResult: SearchResult? = null,
+        // Last batch result will help RecyclerView to position itself
+        val lastBatch: SearchResult? = null,
```

Comment:
Why not using the same mechanism than for `RoomUploadsViewState` with an Async for the current request?

Reply:
Done.